src/vsr/journal.zig

```diff
@@ -1308,20 +1358,16 @@ pub fn Journal(comptime Replica: type, comptime Storage: type) type {
                     self.faulty.clear(slot);
                 },
                 .fix => {
-                    // TODO Perhaps we should have 3 separate branches here for the different cases.
-                    // The header may be valid or invalid.
-                    // The header may be reserved or a prepare.
-                    assert(prepare.?.command == .prepare);
-                    assert(self.prepare_inhabited[slot.index]);
-                    assert(self.prepare_checksums[slot.index] == prepare.?.checksum);
-
                     self.headers[slot.index] = prepare.?.*;
                     self.faulty.clear(slot);
                     if (replica.replica_count == 1) {
-                        // @E, @F, @G, @H, @K:
+                        // @D, @E, @F, @G, @H, @K:
                         self.dirty.clear(slot);
                         // TODO Repair header on disk to restore durability.
                     } else {
+                        assert(prepare.?.command == .prepare);
+                        assert(self.prepare_inhabited[slot.index]);
+                        assert(self.prepare_checksums[slot.index] == prepare.?.checksum);
```

Comment:
Do these not apply to the first branch any longer?

Reply:
Correct, case `@D` will `fix` a reserved header.